docs/src/ref/gfi.md

```diff
@@ -81,54 +93,61 @@ There is also a family of probability distributions ``q(r; x, t)`` on untraced r
 q(r; x, t) > 0 \mbox{ if and only if } p(r; x, t) > 0
 ```
 
+
 ## Traces
 
 An **execution trace** (or just *trace*) is a record of an execution of a generative function.
-There is no abstract type representing all traces.
-Different concrete types of generative functions use different data structures and different Julia types for their traces.
-The trace type that a generative function uses is the second type parameter of the [`GenerativeFunction`](@ref) abstract type.
+Traces are the primary data structures manipulated by Gen inference programs.
+There are various methods for producing, updating, and inspecting traces.
+Traces contain:
 
-A generative function can be executed and traced using:
-```@docs
-simulate
-```
+- the arguments to the generative function
 
-An traced execution that satisfies constraints on the choice map can be generated using:
-```@docs
-generate
-```
+- the random choice map 
 
-The trace contains various information about the execution, including:
+- the return value
 
-The arguments to the generative function:
-```@docs
-get_args
-```
+- auxiliary state
 
-The return value of the generative function:
-```@docs
-get_retval
-```
+- other implementation-specific state that is not exposed to the caller or user of the generative function, but is used internally to facilitate e.g. incremental updates to executions.
```

Comment:
Are you counting non-addressable randomness as one of these?  I think it might need its own item in this list because it's part of the public interface?  Its form is unspecified, and it need not have an accessor, but the public interface to traces uses math that depends on `r`, so every trace type has to have a notion of `r`.  WDYT?

Reply:
The non-addressable randomness itself does not need to be stored. For example in many cases, the only thing that depends on `r` that needs to be stored is the 'weight' `p(r, t; x) / q(r; x, t)` and not `r` itself. But I'll add another bullet for it as you suggested.

Reply:
Fixed in f971f9c930b97c55233839cfe1f71d2741d3c653

Reply:
Right, right -- and we're using the word "contains" loosely here, since the GFI can't say the trace has to actually contain something, it can just say the trace has to have methods that act as if it contained them.  Nice phrasing for the extra bullet point.  Thanks!